### PR TITLE
#164198157 Implements search and filter

### PR DESCRIPTION
**Pull request by valenstical**
#### What does this PR do?
This PR implements a feature that allows user search for articles by article name, author name or by category.

#### Description of Task to be completed

- [x] Write unit test
- [x] Create presentational and container Search components
- [x] Add search container component to the system route
- [x] Create relevant selectors, actions, and reducers for the search component
- [x] Consume API endpoint to search articles
- [x] Write integration tests

#### How should this be manually tested

1. Clone repo by running the following command on your terminal:

```bash
git clone -b feature/164198157/search-and-filter--single-branch \
https://github.com/andela/apollo-ah-frontend.git
```

2. Navigate into the repo directory: `cd apollo-ah-frontend`

2. Install dependency packages with: `npm install`

3. On the root directory, create a `.env` file from `.env.example` and provide required credentials

4. Run `npm test`

#### Any background context you want to provide
N/A

#### What are the relevant pivotal tracker stories

[#164198157](https://www.pivotaltracker.com/story/show/164198157)

#### Screenshots
ezgif-4-6ecb449b6cb8

### CODE REVIEW

File: src/actions/articleAction.js [hunk 1]
```diff
@@ -50,3 +50,28 @@ export const getArticles = (page, size) => async (dispatch) => {
...
+//   return request({ route: 'articles?size=12', }).then((response) => {
+//     const { data: { data: { articles } } } = response;
+//     dispatch(action(getArticlesType.success, articles));
+//   }).catch((error) => {
+//     const message = error.response
+//       ? ''
+//       : 'Please check your network connection';
+//     dispatch(action(getArticlesType.failure, message));
+//   });
+// };
```
Comment (devdbrandy) [inline src/actions/articleAction.js]: **Nitpick**

- I believe these are dead codes, we shouldn't check out commented code into the repository

**Review by devdbrandy — CHANGES_REQUESTED** [review]
Great job @valenstical, we can now search for articles ✨ 💃. I have just one requested change, kindly attend to it, thanks.

**Review by thislekan — APPROVED** [review]
LGTM

File: src/actions/articleAction.js (hunk 1 above)
Comment (valenstical) [reply src/actions/articleAction.js]: Thanks for pointing that out. I have removed the code.

**Review by devdbrandy — APPROVED** [review]

**Comment by devdbrandy** [comment]
Looks good to me ✨ ✨ 👍

**Pull request merged**